### DISCUSSION TITLE
serve site on a different port

### DISCUSSION THREAD

**Discussion by setop** [nuejs/nue#275] [Q&A] [answered]
I tried `nue` to serve my site but I get `!! Failed to start server. Is port 8080 in use? `. Yes it is, and it won't change.

I browsed the documentation to see how to change the port but have not found the information.

**Comment by nobkd** [comment]
Add
`port: <port>`
to your `site.yaml` in the root of your *nue* project.

See https://nuejs.org/docs/reference/configuration-options.html#port

**Reply by setop** [reply]
thanks
it is a nice workaround.
but I'd like it to be part of the CLI options, and not embedded in the site description, which should more business than technical.
else, it is cumbersome to change it from one environment to the other.

**Reply by nobkd** [reply, marked as answer]
There is no command line option for the port, but you can create another environment file that overrides values in `site.yaml` or defaults with its own. This can be useful, if you want to change more than just one value for your setup in dev VS prod. 

See `nue --help` or https://nuejs.org/docs/reference/command-line-interface.html or:

https://github.com/nuejs/nue/blob/b8a542a1c55354a5cfa8d04613205d351748c783/packages/nuekit/src/cli-help.js#L15-L22

---

`site.yaml`
```yaml
port: 8080 # default, can be left out
```

define a (e.g.) `dev.yaml` beside your `site.yaml`

`dev.yaml`
```yaml
port: 8888
```

use with `nue -e dev.yaml` for development and `nue` without the env file for production. the content of the `-e` option can also be a relative path, I think.

This is not tested, but should work.

**Reply by tipiirai** [reply]
A `--port` option will likely be added at some point. Such an expected option.

**Reply by nobkd** [reply]
Port CLI option was implemented in #330.
It can be used with `-P <port>` or `--port <port>`
